src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -49,6 +49,16 @@ class GraphFrame private(
   /** Default constructor is provided to support serialization */
   protected def this() = this(null, null)
 
+  override def toString: String = {
+    // We call select on the vertices and edges to ensure that ID, SRC, DST always come first
+    // in the printed schema.
+    val v = vertices.select(ID, vertices.columns.filter(_ != ID) :_ *).toString
+    val e = edges.select(SRC, DST +: edges.columns.filter(c => c != SRC && c != DST) :_ *).toString
+    "GraphFrame\n" +
```

Comment:
Thanks, looks good to me!

On Sun, Feb 21, 2016 at 2:04 PM, jkbradley <notifications@github.com> wrote:

> In src/main/scala/org/graphframes/GraphFrame.scala
> <https://github.com/graphframes/graphframes/pull/11#discussion_r53574335>:
>
> > @@ -49,6 +49,16 @@ class GraphFrame private(
> >    /** Default constructor is provided to support serialization */
> >    protected def this() = this(null, null)
> >
> > +  override def toString: String = {
> > +    // We call select on the vertices and edges to ensure that ID, SRC, DST always come first
> > +    // in the printed schema.
> > +    val v = vertices.select(ID, vertices.columns.filter(_ != ID) :_ *).toString
> > +    val e = edges.select(SRC, DST +: edges.columns.filter(c => c != SRC && c != DST) :_ *).toString
> > +    "GraphFrame\n" +
>
> done!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/graphframes/graphframes/pull/11/files#r53574335>.
>